File: arrays/Trapping_Rain_Water_Java/trapping_rain_water.java
class TrappingWater {

	public static void main(String[] args){
		int[] arr = {3,0,1,2,5};
		int res = 0;
		for (int i=1;i<arr.length;i++){
			int left = arr[i];
			for (int j=0;j<i;j++){
				left = Math.max(left,arr[j]);
			}
			int right = arr[i];
			for (int j=i+1;j<arr.length;j++){
				right = Math.max(right,arr[j]);
			}
			res=res+(Math.min(left,right)-arr[i]);
		}
		System.out.println("Trapped water is "+res);
	}
}

File: arrays/trapping_rain_water.py
my_list = [0, 1, 0, 2, 1, 0, 1, 3, 2, 1, 2, 1]
res = 0 
for i in range (1,len(my_list)-1):
    left = my_list[i]
    for j in range(0,i):
        left=max(left,my_list[j])
    right = my_list[i]
    for j in range(i+1,len(my_list)-1):
        right=max(right,my_list[j])
    res=res+(min(left,right)-my_list[i])

print ("Trapped water is :{0}".format(res))
